vault/resource_azure_secret_backend_test.go

```diff
@@ -62,6 +62,42 @@ func TestAzureSecretBackend(t *testing.T) {
 			},
 		},
 	})
+	azureCheckFuncs := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
+		resource.TestCheckResourceAttr(resourceName, "tenant_id", "11111111-2222-3333-4444-222222222222"),
+		resource.TestCheckResourceAttr(resourceName, "client_id", "11111111-2222-3333-4444-333333333333"),
+		resource.TestCheckResourceAttr(resourceName, "client_secret", "12345678901234567890"),
+		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
+	}
+
+	skipMSGraphCheck := provider.IsAPISupported(testProvider.Meta(), provider.VaultVersion112)
+	if !skipMSGraphCheck {
+		azureCheckFuncs = append(azureCheckFuncs,
+			resource.TestCheckResourceAttr(resourceName, "use_microsoft_graph_api", "false"))
+	}
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+		},
+		CheckDestroy: testCheckMountDestroyed(resourceType, consts.MountTypeAzure, consts.FieldPath),
+		Steps: []resource.TestStep{
+			{
+				Config: testAzureSecretBackend_initialConfig(path),
```

Comment:
These currently fail because the values we expect do not match up as we are updating the config. I believe you will need checkFuncs for individual cases, something like:

```
	azureInitialCheckFuncs := []resource.TestCheckFunc{
		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
		resource.TestCheckResourceAttr(resourceName, "tenant_id", "11111111-2222-3333-4444-222222222222"),
		resource.TestCheckResourceAttr(resourceName, "client_id", "11111111-2222-3333-4444-333333333333"),
		resource.TestCheckResourceAttr(resourceName, "client_secret", "12345678901234567890"),
		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
	}

	azureUpdatedCheckFuncs := []resource.TestCheckFunc{
		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
		resource.TestCheckResourceAttr(resourceName, "tenant_id", "22222222-3333-4444-5555-333333333333"),
		resource.TestCheckResourceAttr(resourceName, "client_id", "22222222-3333-4444-5555-444444444444"),
		resource.TestCheckResourceAttr(resourceName, "client_secret", "098765432109876543214"),
		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
	}
```

Reply:
The we can match them along with their config:

```
			{
				Config: testAzureSecretBackend_initialConfig(path),
				Check:  resource.ComposeTestCheckFunc(azureInitialCheckFuncs...),
			},
			{
				Config: testAzureSecretBackend_updated(path),
				Check:  resource.ComposeTestCheckFunc(azureUpdatedCheckFuncs...),
			},
```